### PR TITLE
Chore/upgrade-oauth2 - update dependencies to support OAuth2 2.x

### LINKED ISSUE #3057: Chore: Update Oauth2
**Because**
Our current version (1.x) has reached EOL and is no longer supported. 

**Acceptance Criteria**
- [ ] The app is using oauth2 and there are no deprecation warnings

**Notes**

`bundle outdated` can be used to  check gems that are out of date. Adjacent gems (ie GitHub / Google Oauth) should be checked too.

Deprecation warning:
```
Post-install message from oauth2:

You have installed oauth2 version 1.4.10, which is EOL.
No further support is anticipated for the 1.4.x series.

OAuth2 version 2 is released.
There are BREAKING changes, but most will not encounter them, and upgrading should be easy!

Please see:
• https://github.com/oauth-xx/oauth2#what-is-new-for-v20
• https://github.com/oauth-xx/oauth2/blob/master/CHANGELOG.md

Please upgrade, report issues, and support the project! Thanks, |7eter l-|. l3oling
```

### PR DESCRIPTION
**Pull request by JacobMarq**
## Because

Our current version (1.x) has reached EOL and is no longer supported.

## This PR

Upgrades the necessary dependencies to support OAuth2 versions 2.x
- update omniauth-github gem to latest version
- update omniauth-google-oauth gem to latest version
- use updated github_api gem to support OAuth2 because original github_api by piotrmurach currently seems to be abandoned

## Issue

Closes #3057 

## Additional Information

Work regarding this PR took getting in contact with multiple maintainers of the mentioned gemfiles and either making updates myself or requesting them. After reaching out to the maintainers for github_api and several months passing with no response, it seems as though this gem has been abandoned. I took it upon myself to make the simple changes to the repo and have the now updated version hosted on rubygems under [github_api2](https://rubygems.org/gems/github_api2). This gem is now being used in place of github_api.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

**Comment by ChargrilledChook** [comment]
Really appreciate the work you put into the @JacobMarq, LGTM at a glance but will try to have  a closer look in the next 2 - 3 days if no one else does :slightly_smiling_face:

File: Gemfile [hunk 1]
```diff
@@ -17,15 +17,15 @@ gem 'discard', '~> 1.2'
 gem 'discordrb-webhooks', '~> 3.4'
 gem 'dry-initializer', '~> 3.1'
 gem 'friendly_id', '~> 5.5'
-gem 'github_api', '~> 0.19'
+gem 'github_api2', '~> 1.0'
```
Comment (KevinMulhern) [inline Gemfile:20]: Hey @JacobMarq, I was just looking around to see what we use this for, it doesn't look like we use it anymore and we can remove it I'm afraid. 

The Git history says it was once used to pull in the curriculum lesson content. But we must have switched to [Octokit](https://github.com/octokit/octokit.rb) for that at some point and forgot to remove this dependency. 

It's probably not much consolation, but thank you so much for going so far out your way to get this working. Hopefully other projects that use still github_api can find your fork and make use of it.

**Review by KevinMulhern — CHANGES_REQUESTED** [review]

**Comment by KevinMulhern** [comment]
Testing Github oauth and it's working perfectly 🔥 
Will get Google oauth setup on my local machine to test that in the next few days

**Comment by KevinMulhern** [comment]
Google Oauth tested and working great ✅ 

Thanks for all your work on this @JacobMarq  
When you remove `github_api2` from the Gemfile this can be merged 🚀

**Comment by KevinMulhern** [comment]
Hey @JacobMarq, do you still want to get this one over the line?  I don't mean to pester or put any pressure on you but we've been getting a few emails from Google about the oauth endpoint shut down and would like to get this resolved asap.

**Comment by JacobMarq** [comment]
Hey @KevinMulhern, Sorry about that, been completely booked between work and school. I can work on this later today.

**Comment by KevinMulhern** [comment]
Thans @JacobMarq  💪

**Comment by JacobMarq** [comment]
@KevinMulhern Change done and Merge Conflicts fixed 👍

**Review by KevinMulhern — APPROVED** [review]
Thank you for all the effort and time you put into this @JacobMarq, we really appreciate it 🥇

**Pull request merged**